### PR TITLE
task 5149 create ui play music

### PR DESCRIPTION
**Pull request by QNFC**
Create ui play music
Link redmine: https://edu-redmine.framgia.vn/issues/5149

### CODE REVIEW

File: app/src/main/res/layout/activity_play_music.xml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        android:src="@drawable/ic_button_download"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="MissingConstraints"
+        />
+
+    <TextView
+        android:id="@+id/textViewSong"
+        android:layout_width="@dimen/dp_150"
+        android:layout_height="50dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: hardcode

File: app/src/main/res/layout/activity_play_music.xml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        android:textSize="@dimen/sp_20"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/textViewSong"
+        tools:ignore="MissingConstraints"
+        />
+
+    <de.hdodenhof.circleimageview.CircleImageView
+        android:id="@+id/circleImagePoster"
+        android:layout_width="200dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        android:layout_height="250dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        android:layout_marginTop="36dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="@+id/seekBarMusic"
+        />
+
+    <TextView
+        android:id="@+id/textViewFullDuration"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="16dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        app:layout_constraintTop_toTopOf="@+id/seekBarMusic"
+        />
+
+    <SeekBar
+        android:id="@+id/seekBarMusic"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="8dp"
+        android:layout_marginTop="8dp"
+        android:progress="50"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        app:layout_constraintStart_toEndOf="@+id/textViewDuration"
+        app:layout_constraintTop_toTopOf="@+id/guideline"
+        style="@style/MySeekBar"
+        />
+
+
+    <ImageView
+        android:id="@+id/imageViewShuffle"
+        android:layout_width="0dp"
+        android:layout_height="25dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+        app:layout_constraintStart_toEndOf="@+id/imageViewShuffle"
+        app:layout_constraintTop_toTopOf="@+id/imageViewPlay"
+        app:srcCompat="@drawable/ic_button_previous"
+        />
+
+    <ImageButton
+        android:id="@+id/imageViewPlay"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="8dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/java/com/framgia/music_48/screen/playmusic/PlayMusicActivity.java [hunk 9]
```diff
@@ -22,7 +41,37 @@ public static Intent getPlayMusicIntent(Context context, Song song, int position
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_play_music);
+        initView();
+        updateUI();
+    }
+
+    private void updateUI() {
```
Comment (buivanhieu7112) [inline app/src/main/java/com/framgia/music_48/screen/playmusic/PlayMusicActivity.java:55]: tách method ra cho đúng chức năng bạn

File: app/src/main/java/com/framgia/music_48/data/source/remote/fetchdata/SongFromURL.java [hunk 10]
```diff
@@ -31,7 +32,7 @@ public SongFromURL(DataRemoteListener dataRemoteListener) {
             String json = getJSonFromURL(strings[0]);
             songs = readJSonFromURL(json);
         } catch (IOException | JSONException e) {
-            mDataRemoteListener.onFetchSongError(e);
+            mExceptionError =e;
```
Comment (buivanhieu7112) [inline app/src/main/java/com/framgia/music_48/data/source/remote/fetchdata/SongFromURL.java]: reformat code

File: app/src/main/java/com/framgia/music_48/screen/listmusicdetail/ListMusicFragment.java [hunk 11]
```diff
@@ -76,7 +76,7 @@ public void onGetSongsWithGenresSuccess(List<Song> songs) {
 
     @Override
     public void onError(Exception ex) {
-        Toast.makeText(getContext(), "" + ex.getMessage(), Toast.LENGTH_SHORT).show();
+        Toast.makeText(getActivity(), "Please Check Your Connection", Toast.LENGTH_SHORT).show();
```
Comment (buivanhieu7112) [inline app/src/main/java/com/framgia/music_48/screen/listmusicdetail/ListMusicFragment.java]: hardcode

File: app/src/main/res/layout/activity_main.xml [hunk 12]
```diff
@@ -18,6 +19,7 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:background="@drawable/navigator_background"
+            tools:ignore="MissingConstraints"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/activity_main.xml]: remove this line

File: app/src/main/res/layout/activity_play_music.xml [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+    <ImageView
+        android:id="@+id/imageViewBack"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/dp_10"
+        android:layout_marginTop="@dimen/dp_10"
+        android:src="@drawable/ic_button_back"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="MissingConstraints"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+    <ImageView
+        android:id="@+id/imageViewDownload"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/dp_10"
+        android:layout_marginTop="@dimen/dp_10"
+        android:src="@drawable/ic_button_download"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="MissingConstraints"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/activity_play_music.xml [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+        android:text="@string/song"
+        android:gravity="center"
+        app:layout_constraintTop_toTopOf="@id/imageViewDownload"
+        android:textSize="@dimen/sp_20"
+        android:textStyle="bold"
+        android:textColor="@color/color_pink_200"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/toolbarMain"
+        tools:ignore="MissingConstraints"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/activity_play_music.xml]: as above

File: app/src/main/res/layout/item_genres_detail.xml [hunk 16]
```diff
@@ -32,6 +34,7 @@
             android:textColor="@android:color/black"
             android:textSize="@dimen/sp_15"
             app:layout_constraintLeft_toRightOf="@id/imageViewPoster"
+            app:layout_constraintTop_toTopOf=""
```
Comment (QuangHieuu) [inline app/src/main/res/layout/item_genres_detail.xml]: ? bạn đã thử build lên chưa?

**Review by QuangHieuu — APPROVED** [review]

**Comment by QNFC** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

File: app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -16,4 +16,7 @@
     <string name="classical">classical</string>
     <string name="country">country</string>
     <string name="no_internet">No Internet</string>
+    <string name="_00_00">00:00</string>
```
Comment (phuongohn-1626) [inline app/src/main/res/values/strings.xml:19]: rename to everyone know

**Pull request merged**